### ISSUE TITLE
C07:Tagestemperatur_HK1 missing

### ISSUE THREAD

**Issue by Fl77777** [GollmerSt/SolvisSmartHomeServer#14]
the Topic ...C07:Tagestemperatur_HK1 is missing and this would be the most important parameter for me to manipulate.
...C13:Tagestemperatur_HK2 is recognized and can also be manipulated.
 My configuration of the base.xml is: 		
<tns:Unit id="mySolvis" type="SolvisMax6"
mainHeating="Extern" heatingCircuits="2" fwLth2_21_02A="false"
Learning is done without any error messages with Configuration mask: 0x100002b
Any ideas?

**Comment by Fl77777** [comment]
I think there might be a mistake in control.xml in line 2644 that becomes appearent with a heating system with 2 heating circuits:

				<tns:ConfigurationMask andMask="0x07"
					compareMask="0x01" />

It should be 0x01 instead of 0x07. Otherwise C07 is not set in a 2 circuit heating system. Changing it to 0x01 solved the issue for me.

**Comment by GollmerSt** [comment]
I have to reopen this issue because I don't understand the background of this problem. The workaround above may not be entirely correct.

When developing the SolvisSmartHomeServer, I only had one system with one heating circuit available.
I was then able to evaluate the program with a user using a system with 3 heating circuits. I haven't been able to test it with two heating circuits yet.

In the case of a system with 3 heating circuits, the home screen looks like this:

![Screenshot](https://github.com/GollmerSt/SolvisSmartHomeServer/blob/master/docu/images/LearnedImages%20mit%20Varianten/001_Home%203%20HK%20ohne%20Innenthermostat.png)

What does the home screen look like for a system with 2 stations?

**Comment by Fl77777** [comment]
Hi, the home screen with two heating circuits looks like this.
![Home screen](https://github.com/user-attachments/assets/7c62c549-00bd-4e8e-909e-79423cc479ed)

**Comment by GollmerSt** [comment]
I've now looked at the configuration file again. You were right, your change should solve the problem, thank you very much. For the other settings - like night temperature - I already had it correct in the configuration.xml.

I will provide a fixed package in the next few days. This will be the only change in the package.

**Comment by Fl77777** [comment]
Hi, 
Thank you very much. I would like to mention that I also had to change this section (lines 3014 and 3015 in control.xml)


<tns:ChannelDescription id="C17" unit="°C">
<tns:Configuration>
<tns:ConfigurationMask andMask="0x04" 
       compareMask="0x04"/>

to 

tns:ConfigurationMask andMask="0x01" 
     compareMask="0x01"/>

for my system to recognize C17 as an adjustable parameter.
Whilst you are at it, could you maybe have a look if this change is correct for any heating system configuration.

Thank you!

**Comment by GollmerSt** [comment]
This is also a bug, Vorlauf_Soll_HK2 would only be adjustable with 3 heating circuits.

From my point of view, the following setting would be correct:

<tns:Configuration>
    <tns:ConfigurationMask andMask="0x02"
            compareMask="0x02" />
</tns:Configuration>

Otherwise this would also be carried out for one heating circuit.

Could you try this entry?

Many thanks.

**Comment by Fl77777** [comment]
You are right, 0x01 would  not be correct for a system with only one heating circuit. For my system with 2 heating circuits (logical and with 1011) it also works with 
<tns:ConfigurationMask andMask="0x02"
compareMask="0x02" />
Thank you very much. Its a great interface and has been working very reliably once I had fixed these two little issues.
